kitsune/wiki/views.py

```diff
@@ -455,7 +455,89 @@ def steal_lock(request, document_slug, revision_id=None):
+    """Create a new revision of a wiki document"""
+    user = request.user
+
+    doc = get_visible_document_or_404(
+        user,
+        locale=request.LANGUAGE_CODE,
+        slug=document_slug,
+        look_for_translation_via_parent=True,
+        return_parent_if_no_translation=True,
+    )
+
+    if doc.locale != request.LANGUAGE_CODE:
+        # We've fallen back to the parent, since no visible translation existed.
+        url = reverse("wiki.translate", locale=request.LANGUAGE_CODE, args=[document_slug])
+        return HttpResponseRedirect(url)
+
+    # If this document has a parent, then the edit is handled by the
+    # translate view. Pass it on.
+    if doc.parent:
+        return translate(request, doc.parent.slug, revision_id)
+    if revision_id:
+        rev = get_object_or_404(Revision, pk=revision_id, document=doc)
+    else:
+        rev = doc.current_revision or doc.revisions.order_by("-created", "-id")[0]
+
+    rev_form = None
+
+    if request.method == "GET":
+        if doc.allows(user, "create_revision"):
+            rev_form = RevisionForm(instance=rev, initial={"based_on": rev.id, "comment": ""})
+            show_revision_warning = _show_revision_warning(doc, rev)
+            locked, locked_by = _document_lock(doc.id, user.username)
+
+            return render(
+                request,
+                "wiki/edit.html",
+                {
+                    "revision_form": rev_form,
+                    "document": doc,
+                    "show_revision_warning": show_revision_warning,
+                    "locked": locked,
+                    "locked_by": locked_by,
+                },
+            )
+        # You can't do anything on this page, so get lost.
+        raise PermissionDenied
+
+    else:  # POST
+        _document_lock_clear(doc.id, user.username)
+
+        if doc.allows(user, "create_revision"):
+            rev_form = RevisionForm(request.POST)
+            rev_form.instance.document = doc  # for rev_form.clean()
+            if rev_form.is_valid():
+                _save_rev_and_notify(rev_form, user, doc, base_rev=rev)
+                if "notify-future-changes" in request.POST:
+                    EditDocumentEvent.notify(request.user, doc)
+                return HttpResponseRedirect(
+                    reverse("wiki.document_revisions", args=[document_slug])
+                )
+        else:
+            raise PermissionDenied
+
+    show_revision_warning = _show_revision_warning(doc, rev)
+    locked, locked_by = _document_lock(doc.id, user.username)
+
+    return render(
+        request,
+        "wiki/edit.html",
+        {
+            "revision_form": rev_form,
+            "document": doc,
+            "show_revision_warning": show_revision_warning,
+            "locked": locked,
+            "locked_by": locked_by,
+        },
+    )
+
+
+@require_http_methods(["GET", "POST"])
+@login_required
+def edit_document_doc(request, document_slug, revision_id=None):
```

Comment:
What do you think of `edit_document_metadata` instead?

Reply:
It was difficult to know where to put an alternative for this view, so I'll add my suggestion here:
```python
@require_http_methods(["GET", "POST"])
@login_required
def edit_document_metadata(request, document_slug, revision_id=None):
    """Edit document metadata."""
    user = request.user

    doc = get_visible_document_or_404(
        user,
        locale=request.LANGUAGE_CODE,
        slug=document_slug,
        look_for_translation_via_parent=True,
        return_parent_if_no_translation=True,
    )

    if doc.locale != request.LANGUAGE_CODE:
        # We've fallen back to the parent, since no visible translation existed.
        url = reverse("wiki.translate", locale=request.LANGUAGE_CODE, args=[document_slug])
        return HttpResponseRedirect(url)

    # If this document has a parent, then the edit is handled by the
    # translate view. Pass it on.
    if doc.parent:
        return translate(request, doc.parent.slug, revision_id)

    if not doc.allows(user, "edit") and user.is_staff:
        raise PermissionDenied

    can_edit_needs_change = doc.allows(user, "edit_needs_change")
    can_archive = doc.allows(user, "archive")

    if revision_id:
        rev = get_object_or_404(Revision, pk=revision_id, document=doc)
    else:
        rev = doc.current_revision or doc.revisions.order_by("-created", "-id")[0]

    if request.method == "GET":
        doc_form = DocumentForm(
            initial=_document_form_initial(doc),
            can_archive=can_archive,
            can_edit_needs_change=can_edit_needs_change,
        )

    else:  # POST
        _document_lock_clear(doc.id, user.username)

        post_data = request.POST.copy()
        post_data.update({"locale": request.LANGUAGE_CODE})

        topics = []
        for t in post_data.getlist("topics"):
            topics.append(int(t))
        post_data.setlist("topics", topics)

        doc_form = DocumentForm(
            post_data,
            instance=doc,
            can_archive=can_archive,
            can_edit_needs_change=can_edit_needs_change,
        )
        if doc_form.is_valid():
            # Get the possibly new slug for the imminent redirection:
            try:
                doc = doc_form.save(None)
            except (TitleCollision, SlugCollision) as e:
                # TODO: .add_error() when we upgrade to Django 1.7
                errors = doc_form._errors.setdefault("title", ErrorList())
                message = "The {type} you selected is already in use."
                message = message.format(
                    type="title" if isinstance(e, TitleCollision) else "slug"
                )
                errors.append(_(message))
            else:
                # Do we need to rebuild the KB?
                _maybe_schedule_rebuild(doc_form)

                return HttpResponseRedirect(
                    urlparams(reverse("wiki.document", args=[doc.slug]))
                )

    show_revision_warning = _show_revision_warning(doc, rev)
    locked, locked_by = _document_lock(doc.id, user.username)

    return render(
        request,
        "wiki/edit_metadata.html",
        {
            "document_form": doc_form,
            "document": doc,
            "show_revision_warning": show_revision_warning,
            "locked": locked,
            "locked_by": locked_by,
        },
    )
```

Reply:
I'm going to go ahead and include the change.